### PR TITLE
channel: support set optimization target

### PR DESCRIPTION
**Pull request by BusyJay**
From 1.6.0 forward, grpc support setting an optimization target for channel, which may be helpful for tuning performance.

### CODE REVIEW

File: src/channel.rs [hunk 1]
```diff
@@ -267,12 +306,44 @@ impl ChannelBuilder {
...
+        self.options.insert(
+            Cow::Borrowed(OPT_OPTIMIZATION_TARGET),
+            Options::String(val.unwrap()),
+        );
+        self
+    }
+
+    /// Set a raw int configuration.
+    #[doc(hidden)]
+    pub fn raw_cfg_int(mut self, key: CString, val: usize) -> ChannelBuilder {
```
Comment (overvenus) [inline src/channel.rs]: Why `#[doc(hidden)]`? I think it would be convenient for users to testing new configuration.

File: src/channel.rs [hunk 2]
```diff
@@ -77,10 +79,20 @@ enum Options {
...
+    Blend,
+    /// Maximize throughput at the expense of latency.
+    Throughput,
+}
+
 /// Channel configuration object.
 pub struct ChannelBuilder {
     env: Arc<Environment>,
-    options: HashMap<&'static [u8], Options>,
+    options: HashMap<Cow<'static, [u8]>, Options>,
```
Comment (siddontang) [inline src/channel.rs]: why use Cow here?

Comment (BusyJay) [reply src/channel.rs]: Because both `&[u8]` and `CString` is accepted here.

File: src/channel.rs (hunk 1 above)
Comment (BusyJay) [reply src/channel.rs]: This method is only for test purpose, users should use the encapsulated one instead.

File: src/channel.rs (hunk 2 above)
Comment (siddontang) [reply src/channel.rs]: I think we can define three static &u[8] for OptTarget, so we don't need CString here.

Comment (BusyJay) [reply src/channel.rs]: `CString` is required by `Options`.

Comment (siddontang) [reply src/channel.rs]: But I see all your old options don't need it.

Comment (BusyJay) [reply src/channel.rs]: See L111, L185 and L204.

Comment (siddontang) [reply src/channel.rs]: For L111, seem we still can use static u[8] as the key.

File: src/channel.rs (hunk 1 above)
Comment (siddontang) [reply src/channel.rs]: Please add a comment for why we need these raw functions.

**Pull request merged**